### ISSUE TITLE
PV internals are written to SQLite

### ISSUE THREAD

**Issue by FGasper** [DBD-SQLite/DBD-SQLite#66]
```
> perl -MDBI -e'my $epee = "épée"; utf8::upgrade($epee); my $dbh = DBI->connect("dbi:SQLite:dbname=mysqlite","",""); $dbh->do("CREATE TABLE foo (val str)"); $dbh->do("INSERT INTO foo VALUES (?)", undef, $epee)'

> echo 'select * from foo' | sqlite3 mysqlite
Ã©pÃ©e
```
^^ In the above code, shouldn’t `épée` be what’s written to the SQLite DB?

**Comment by charsbar** [comment]
Sorry for the late reply.

EXAMPLE 1: with sqlite_unicode => 1, utf8::upgrade
```
$ perl -MDBI -E'unlink "mysqlite"; my $epee = "épée"; utf8::upgrade($epee); my $dbh = DBI->connect("dbi:SQLite:dbname=mysqlite","","",{sqlite_unicode =>1}); $dbh->do("CREATE TABLE foo (val str)"); $dbh->do("INSERT INTO foo VALUES (?)", undef, $epee); say join ",", $dbh->selectrow_array("SELECT * FROM foo")'
épée
$ echo 'select * from foo' | sqlite3 mysqlite
Ã©pÃ©e
```
EXAMPLE 2: with sqlite_unicode => 0, utf8::upgrade (as your first comment)
```
$ perl -MDBI -E'unlink "mysqlite"; my $epee = "épée"; utf8::upgrade($epee); my $dbh = DBI->connect("dbi:SQLite:dbname=mysqlite","","",{sqlite_unicode =>0}); $dbh->do("CREATE TABLE foo (val str)"); $dbh->do("INSERT INTO foo VALUES (?)", undef, $epee); say join ",", $dbh->selectrow_array("SELECT * FROM foo")'
Ã©pÃ©e
$ echo 'select * from foo' | sqlite3 mysqlite
Ã©pÃ©e
```
EXAMPLE 3: with sqlite_unicode => 1, without utf8::upgrade
```
$ perl -MDBI -E'unlink "mysqlite"; my $epee = "épée";  my $dbh = DBI->connect("dbi:SQLite:dbname=mysqlite","","",{sqlite_unicode =>1}); $dbh->do("CREATE TABLE foo (val str)"); $dbh->do("INSERT INTO foo VALUES (?)", undef, $epee); say join ",", $dbh->selectrow_array("SELECT * FROM foo")'
épée
$ echo 'select * from foo' | sqlite3 mysqlite
Ã©pÃ©e
```
EXAMPLE 4: wIth sqlite_unicode => 0, without utf8::upgrade
```
$ perl -MDBI -E'unlink "mysqlite"; my $epee = "épée";  my $dbh = DBI->connect("dbi:SQLite:dbname=mysqlite","","",{sqlite_unicode =>0}); $dbh->do("CREATE TABLE foo (val str)"); $dbh->do("INSERT INTO foo VALUES (?)", undef, $epee); say join ",", $dbh->selectrow_array("SELECT * FROM foo")'
épée
$ echo 'select * from foo' | sqlite3 mysqlite
épée
```
I'm not sure which of the above examples suits your need best, but apparently your code in the first comment is not a good choice.

**Comment by FGasper** [comment]
My point is that utf8::upgrade() doesn’t change a scalar’s string value; thus, it should not affect how Perl translates a scalar to C `char *`.

Did you ever consider use of SvPVbyte insted of SvPV?

**Comment by charsbar** [comment]
Closed as #67 has been merged. Thanks.
